### PR TITLE
Crayon making part 1

### PR DESCRIPTION
**Pull request by jughu**
CHEMIST BEING ABLE TO MAKE THEIR OWN CRAYONS

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)its the tiny details that make the game great

### CODE REVIEW

**Comment by QualityVan** [comment]
nice changelog

**Comment by jughu** [comment]
thank

**Comment by QualityVan** [comment]
also: the reagent_contents list for crayons determines the ratio in which their reagents datum is filled, not the amount by which their reagents datum is filled.  nutriment = 1, powder = 1 means that they'll have equal amounts of nutriment and crayon powder, not that they'll have one of each.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 1]
```diff
@@ -86,6 +88,126 @@
...
+	name = "rainbow Crayonification"
+	id = "crayonwrai"
+	required_reagents = list("lye" =10, "colorful_reagent" = 10)
+	required_temp = 374
+	mob_react = FALSE
+	
+/datum/chemical_reaction/crayonificationrai/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
+		new /obj/item/toy/crayon/rainbow(location)
```
Comment (DoctoreMad) [inline code/modules/reagents/chemistry/recipes/others.dm]: Ripo chef. Nutriment is useless now. Just eat rainbow crayon forever.

Comment (DoctoreMad) [reply code/modules/reagents/chemistry/recipes/others.dm]: Also Is there any reason this section is blue whereas other lines above are color coded?(Ex:/datum/ is always in orange)
Its a useless question I'm not any less curious knowing that.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 2]
```diff
@@ -86,6 +88,126 @@
...
+	
+/datum/chemical_reaction/crayonicationwhite/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
+		new /obj/item/toy/crayon/white(location)
+	
+/datum/chemical_reaction/crayonicationmime
+	name = "Mime Crayon"
+	id = "mimecrayonitation"
+	required_reagents = list("lye" =10, "whitecrayonpowder" = 10, "blackcrayonpowder =10)
```
Comment (deathride58) [inline code/modules/reagents/chemistry/recipes/others.dm]: you forgot the quote at the end of black crayon powder here

**Comment by jughu** [comment]
@DoctoreMad was because the missed quatation that @deathride58 
pointed out

**Comment by fludd12** [comment]
Why not make it with honeycomb or something? Bee wax crayons.

**Comment by jughu** [comment]
because part 2 will involve botany

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 3]
```diff
@@ -87,6 +89,126 @@
 	for(var/i = 1, i <= created_volume, i++)
 		new /obj/item/soap/homemade(location)
 
+/datum/chemical_reaction/crayonicationred
+	name = "Red Crayon"
+	id = "redcrayonication"
+	required_reagents = list("lye" =10, "redcrayonpowder" = 10)
```
Comment (ma44) [inline code/modules/reagents/chemistry/recipes/others.dm]: >no space

Comment (jughu) [reply code/modules/reagents/chemistry/recipes/others.dm]: What no space.

Comment (DoctoreMad) [reply code/modules/reagents/chemistry/recipes/others.dm]: "lye" **=10**

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 4]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationwhite/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
+		new /obj/item/toy/crayon/white(location)
+
+/datum/chemical_reaction/crayonicationmime
+	name = "Mime Crayon"
+	id = "mimecrayonitation"
+	required_reagents = list("lye" = 10, "whitecrayonpowder" = 10, "blackcrayonpowder" =10)
```
Comment (XDTM) [inline code/modules/reagents/chemistry/recipes/others.dm]: the black or white reaction will happen before this, making this impossible to achieve; maybe make it 5 and 5

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/recipes/others.dm]: replace lye with nothing

File: code/modules/reagents/chemistry/recipes/others.dm (hunk 1 above)
Comment (jughu) [reply code/modules/reagents/chemistry/recipes/others.dm]: Rainbow will be more harderl to make in part 2

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 5]
```diff
@@ -74,6 +74,8 @@
 	id = "capsaicincondensation"
 	results = list("condensedcapsaicin" = 5)
 	required_reagents = list("capsaicin" = 1, "ethanol" = 5)
+	
+////////////////////////////////// Lye recipes ///////////////////////////////////
```
Comment (Supermichael777) [inline code/modules/reagents/chemistry/recipes/others.dm]: You do know lye is the "erase all life" bit of soap and not the "water soluble" bit of soap right?

**Comment by Artemiuz28** [comment]
...Why use lye though? Lye is a base, and with fat it kinda makes soap... But crayons? Just do something with this

**Comment by jughu** [comment]
i will in part 2 just WAit

**Comment by jughu** [comment]
UH HELLO SIR THIS IS READY

**Comment by jughu** [comment]
Funny bot.

**Comment by FuryMcFlurry** [comment]
chem trails by drawing dicks next to an airlock. wake up sheeple

**Comment by AnonymousNow** [comment]
Out of interest, does this use the colour of the liquid chemical as seen in beakers/bottles? Hell, can it use any liquid?

"Blood runes, captain."
"No, they're crayon, genius."
"Sir, the crayon was made from blood."
"Ohhhh."

**Comment by jughu** [comment]
Not yet this is part 1 which just covers basic crayons

**Comment by jughu** [comment]
Part 2 will include color powder making and then part 3 will be custum crayons eek

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 6]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationorange
+	name = "Orange Crayon"
+	id = "orangecrayonication"
+	required_reagents = list("wax" = 10, "orangecrayonpowder" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationorange/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 7]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationyellow
+	name = "Yellow Crayon"
+	id = "yellowcrayonication"
+	required_reagents = list("wax" = 10, "yellowcrayonpowder" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationyellow/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 8]
```diff
@@ -87,6 +89,126 @@
...
 
+/datum/chemical_reaction/crayonicationred
+	name = "Red Crayon"
+	id = "redcrayonication"
+	required_reagents = list("wax" = 10, "redcrayonpowder" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationred/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 9]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationblue
+	name = "Blue Crayon"
+	id = "bluecrayonication"
+	required_reagents = list("wax" = 10, "bluecrayonpowder" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationblue/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 10]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationpurple
+	name = "Purple Crayon"
+	id = "purplecrayonication"
+	required_reagents = list("wax" = 10, "purplecrayonpowder" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationpurple/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 11]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationblack
+	name = "Black Crayon"
+	id = "blackcrayonication"
+	required_reagents = list("wax" = 10, "blackcrayonpowder" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationblack/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 12]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationwhite
+	name = "White Crayon"
+	id = "whitecrayonication"
+	required_reagents = list("wax" = 10, "whitecrayonpowder" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationwhite/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 13]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationmime
+	name = "Mime Crayon"
+	id = "mimecrayonitation"
+	required_reagents = list("wax" = 10, "whitecrayonpowder" = 5, "blackcrayonpowder" = 5)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationmime/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 14]
```diff
@@ -87,6 +89,126 @@
...
+
+/datum/chemical_reaction/crayonicationrainbow
+	name = "rainbow Crayon"
+	id = "rainbowcrayonication"
+	required_reagents = list("wax" = 10, "colorful_reagent" = 10)
+	required_temp = 374
+	mob_react = FALSE
+
+/datum/chemical_reaction/crayonicationrainbow/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: location = holder.my_atom.drop_location()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
How is this useful if the only way to get crayon powder is from crayons? 🤔

**Comment by Cyberboss** [comment]
Also add a changelog

**Comment by jughu** [comment]
this is just part 1 cyberboss part 2 will cover powders

**Comment by jughu** [comment]
Going on a week vacation will continue when back

**Pull request closed without merging**